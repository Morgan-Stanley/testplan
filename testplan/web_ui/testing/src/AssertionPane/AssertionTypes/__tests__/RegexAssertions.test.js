import React from "react";
import { shallow } from "enzyme";
import { StyleSheetTestUtils } from "aphrodite";
import { RegexAssertion, RegexMatchLineAssertion } from "../RegexAssertions";

const RegexMatchInputs = [
  {
    assertion: {
      category: "DEFAULT",
      machine_time: "2019-02-12T17:41:43.145337+00:00",
      description: "string pattern match",
      pattern: "foo",
      line_no: 196,
      meta_type: "assertion",
      passed: true,
      match_indexes: [[0, 3]],
      type: "RegexMatch",
      utc_time: "2019-02-12T17:41:43.145332+00:00",
      string: "foobar",
    },
  },
  {
    assertion: {
      category: "DEFAULT",
      machine_time: "2019-02-12T17:41:43.150042+00:00",
      description: null,
      pattern: "baz",
      line_no: 217,
      meta_type: "assertion",
      passed: true,
      match_indexes: [],
      type: "RegexMatchNotExists",
      utc_time: "2019-02-12T17:41:43.150037+00:00",
      string: "foobar",
    },
  },
  {
    assertion: {
      category: "DEFAULT",
      machine_time: "2019-02-12T17:41:43.153301+00:00",
      description: null,
      pattern: "second",
      line_no: 225,
      meta_type: "assertion",
      passed: true,
      match_indexes: [[11, 17]],
      type: "RegexSearch",
      utc_time: "2019-02-12T17:41:43.153294+00:00",
      string: "first line\nsecond line\nthird line",
    },
  },
  {
    assertion: {
      category: "DEFAULT",
      machine_time: "2019-02-12T17:41:43.154688+00:00",
      description: "Passing search empty",
      pattern: "foobar",
      line_no: 230,
      meta_type: "assertion",
      passed: true,
      match_indexes: [],
      type: "RegexSearchNotExists",
      utc_time: "2019-02-12T17:41:43.154681+00:00",
      string: "first line\nsecond line\nthird line",
    },
  },
  {
    assertion: {
      category: "DEFAULT",
      machine_time: "2019-02-12T17:41:43.157637+00:00",
      description: null,
      string: "foo foo foo bar bar foo bar",
      pattern: "foo",
      line_no: 243,
      condition_match: true,
      meta_type: "assertion",
      condition: "<lambda>",
      passed: true,
      type: "RegexFindIter",
      utc_time: "2019-02-12T17:41:43.157630+00:00",
      match_indexes: [
        [0, 3],
        [4, 7],
        [8, 11],
        [20, 23],
      ],
    },
  },
];

describe("RegexAssertions", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it.each(RegexMatchInputs)("shallow renders $assertion.type", (props) => {
    const shallowComponent = shallow(<RegexAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });

  it("shallow renders RegexMatchLineAssertion", () => {
    const props = {
      assertion: {
        category: "DEFAULT",
        machine_time: "2019-02-12T17:41:43.160987+00:00",
        description: null,
        string: "first line\nsecond line\nthird line",
        pattern: "\\w+ line$",
        line_no: 257,
        meta_type: "assertion",
        passed: true,
        type: "RegexMatchLine",
        utc_time: "2019-02-12T17:41:43.160981+00:00",
        match_indexes: [
          [0, 0, 10],
          [1, 0, 11],
          [2, 0, 10],
        ],
      },
    };
    const shallowComponent = shallow(<RegexMatchLineAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});
