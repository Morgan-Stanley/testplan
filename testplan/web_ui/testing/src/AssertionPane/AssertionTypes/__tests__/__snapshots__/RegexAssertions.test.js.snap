// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegexAssertions shallow renders RegexFindIter 1`] = `
<RegexBasedAssertion
  pattern="foo"
>
  <pre
    className="inputText_197f7f1"
  >
    <span
      key="-16494850670,30"
    />
    <mark
      className="mark_vslfl2"
      key="-16494850670,31"
    >
      foo
    </mark>
    <span
      key="-16494850674,70"
    >
       
    </span>
    <mark
      className="mark_vslfl2"
      key="-16494850674,71"
    >
      foo
    </mark>
    <span
      key="-16494850678,110"
    >
       
    </span>
    <mark
      className="mark_vslfl2"
      key="-16494850678,111"
    >
      foo
    </mark>
    <span
      key="-164948506720,230"
    >
       bar bar 
    </span>
    <mark
      className="mark_vslfl2"
      key="-164948506720,231"
    >
      foo
    </mark>
    <span
      key="-164948506723"
    >
       bar
    </span>
  </pre>
</RegexBasedAssertion>
`;

exports[`RegexAssertions shallow renders RegexMatch 1`] = `
<RegexBasedAssertion
  pattern="foo"
>
  <pre
    className="inputText_197f7f1"
  >
    <span
      key="-19797186360,30"
    />
    <mark
      className="mark_vslfl2"
      key="-19797186360,31"
    >
      foo
    </mark>
    <span
      key="-19797186363"
    >
      bar
    </span>
  </pre>
</RegexBasedAssertion>
`;

exports[`RegexAssertions shallow renders RegexMatchLineAssertion 1`] = `
<RegexBasedAssertion
  pattern="\\\\w+ line$"
>
  <pre
    className="inputText_197f7f1"
  >
    <mark
      className="mark_vslfl2"
      key="0,0,10"
    >
      first line

    </mark>
    <mark
      className="mark_vslfl2"
      key="1,0,11"
    >
      second line

    </mark>
    <mark
      className="mark_vslfl2"
      key="2,0,10"
    >
      third line

    </mark>
  </pre>
</RegexBasedAssertion>
`;

exports[`RegexAssertions shallow renders RegexMatchNotExists 1`] = `
<RegexBasedAssertion
  pattern="baz"
>
  <pre
    className="inputText_197f7f1"
  >
    <span
      key="-13621270830"
    >
      foobar
    </span>
  </pre>
</RegexBasedAssertion>
`;

exports[`RegexAssertions shallow renders RegexSearch 1`] = `
<RegexBasedAssertion
  pattern="second"
>
  <pre
    className="inputText_197f7f1"
  >
    <span
      key="-173699572811,170"
    >
      first line

    </span>
    <mark
      className="mark_vslfl2"
      key="-173699572811,171"
    >
      second
    </mark>
    <span
      key="-173699572817"
    >
       line
third line
    </span>
  </pre>
</RegexBasedAssertion>
`;

exports[`RegexAssertions shallow renders RegexSearchNotExists 1`] = `
<RegexBasedAssertion
  pattern="foobar"
>
  <pre
    className="inputText_197f7f1"
  >
    <span
      key="-2053333250"
    >
      first line
second line
third line
    </span>
  </pre>
</RegexBasedAssertion>
`;
